scripts/workers/code_upload_submission_worker.py

```diff
@@ -276,47 +276,62 @@ def update_failed_jobs_and_send_logs(
+                        try:
+                            pod_log_response = (
+                                core_v1_api_instance.read_namespaced_pod_log(
+                                    name=pod_name,
+                                    namespace="default",
+                                    _return_http_data_only=True,
+                                    _preload_content=False,
+                                    container="agent",
+                                )
                             )
-                        )
-                        pod_log = pod_log_response.data.decode("utf-8")
-                        submission_data = {
-                            "challenge_phase": phase_pk,
-                            "submission": submission_pk,
-                            "stdout": "",
-                            "stderr": pod_log,
-                            "submission_status": "FAILED",
-                            "result": "[]",
-                            "metadata": "",
-                        }
-                        response = evalai.update_submission_data(
-                            submission_data, challenge_pk, phase_pk
-                        )
-                        print(response)
-                    except client.rest.ApiException as e:
-                        logger.exception(
-                            "Exception while reading Job logs {}".format(e)
-                        )
+                            pod_log = pod_log_response.data.decode("utf-8")
+                            submission_data = {
+                                "challenge_phase": phase_pk,
+                                "submission": submission_pk,
+                                "stdout": "",
+                                "stderr": pod_log,
+                                "submission_status": "FAILED",
+                                "result": "[]",
+                                "metadata": "",
+                            }
+                            response = evalai.update_submission_data(
+                                submission_data, challenge_pk, phase_pk
+                            )
+                            print(response)
+                        except client.rest.ApiException as e:
+                            logger.exception(
+                                "Exception while reading Job logs {}".format(e)
+                            )
+    except Exception as e:
+        logger.exception("Exception while reading Job {}".format(e))
+        submission_data = {
+            "challenge_phase": phase_pk,
+            "submission": submission_pk,
+            "stdout": "",
+            "stderr": "Submission Job failed.",
+            "submission_status": "failed",
+            "result": "[]",
+            "metadata": "",
+        }
+        response = evalai.update_submission_data(
+            submission_data, challenge_pk, phase_pk
+        )
```

Comment:
@savish28 the submission message also needs to be deleted if the job was deleted on the cluster.

Reply:
@Ram81 Sure, Done.